databaseManager.class.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+    include_once 'databaseManagerInterface.php';
+
+    /**
+     * Define an Abstract class for all databases
+     * connections to extend
+     */
+    abstract class DatabaseManager implements DatabaseManagerInterface
+    {
+        protected $dbDriver;
+        protected $host;
+        protected $port;
+        protected $username;
+        protected $password;
+        protected $dbName;
+        protected $unixSocket;
+        protected $charset;
+        protected $connection;
+
+        /**
+         * Connect database
+         */
+        abstract protected function openConnection();
+
+        /**
+         * Returns rows from the database based on the conditions
+         * @param string name of the table
+         * @param array select, where, order_by, limit and return_type conditions
+         */
+        abstract protected function selectData($table, $conditions = array());
+
+        /**
+         * Insert data into the database
+         * @param string name of the table
+         * @param array the data for inserting into the table
+         */
+        abstract protected function insertData($table,$data);
+        
+        /**
+         * Update data into the database
+         * @param string name of the table
+         * @param array the data for updating into the table
+         * @param array where condition on updating data
+         */
+        abstract protected function updateData($table, $data, $conditions);
+
+        /**
+         * Delete data from the database
+         * @param string name of the table
+         * @param array where condition on deleting data
+         */
+        abstract protected function deleteData($table, $conditions);
+
+        /**
+         * Constructor
+         */
+        protected function __construct($dbDriver, $host, $port, $username, $password, $dbName, $unixSocket, $charset)
+        {
+            $this->dbDriver = $dbDriver;
+            $this->host = $host;
+            $this->port = $port;
+            $this->username = $username;
+            $this->password = $password;
+            $this->dbName = $dbName;
+            $this->unixSocket = $unixSocket;
+            $this->charset = $charset;
+        }
+
+        /**
+         * @return bool
+         */
+        public function isConnected()
+        {
+            return ($this->connection instanceof PDO);
+        }
+
+        /**
+         * Kill connection
+         */
+        public function closeConnection()
+        {
+            $this->connection = null;
```

Comment:
@lumakos By nulling the `connection` property, would close the connection with the database server?

Reply:
The connection remains active for the lifetime of that PDO object. I assign NULL to the variable that holds the object to close the connection, so I destroy the object by ensuring that all remaining references to it are deleted.